kong/plugins/response-transformer/header_transformer.lua

```diff
@@ -1,13 +1,22 @@
 local isempty = require "table.isempty"
-
+local mime_type = require "kong.tools.mime_type"
+local ngx_re = require "ngx.re"
+local pl_stringx = require "pl.stringx"
 
 local kong = kong
 local type = type
-local find = string.find
-local lower = string.lower
 local match = string.match
 local noop = function() end
+local ipairs = ipairs
+local parse_mime_type = mime_type.parse_mime_type
+local mime_type_includes = mime_type.includes
+local split = ngx_re.split
+local strip = pl_stringx.strip
 
+local JSON_MEDIA_TYPES = {
+  { type = "application", subtype = "json" },
+  { type = "application", subtype = "*+json" },
+}
```

Comment:
Is it really necessary to specify that a subtype is allowed?  When parsing a document of a certain media type, its structured subtypes will always be possible.  The subtype information is useful for processors that know how to deal specifically with documents of that subtype.  All other process can always treat all subtype documents as if they were of the main type.  Effectively, when matching a media type without a subtype, the subtype information can always be ignored and the additional `*+json` media type is not required.

Reply:
It depends on how we design the `includes` util function. Here, the `includes` is not taking `application/problem+json` as included to `application/json`, since the last one does not contain the wildcard.

The utils function does not have the knowledge of which specific mime types belong to its “parent” type. It just does the wildcard match.

Reply:
> The utils function does not have the knowledge of which specific mime types belong to its “parent” type. It just does the wildcard match.

As the function is designed to work with mime types, it should know that`application/json` includes `application/problem+json`.

Reply:
If that logic is reused by many places, I think it makes sense to come up with another convenient function that can consider that `application/json`  includes `application/problem+json`.